blocklylib-core/src/main/java/com/google/blockly/model/BlocklyCategory.java

```diff
@@ -371,6 +374,11 @@ public BlockItem(Block block) {
             mBlock = block;
         }
 
+        public BlockItem(Block block, String workspaceId) {
+            this(block);
+            mBlock.setEventWorkspaceId(workspaceId);
```

Comment:
This seems strange to be part of this constructor. First, it is not yet part of a BlockCategory at this point. And second, there might be necessary work to unset it from the previous workspace, if any.

Despite duplicating the call in many places, I would still move this out.

It might make sense to stash the "workspace" id in the BlockCategory, to reference upon add.